### PR TITLE
Compress the series chunk details in index.

### PR DESCRIPTION
**Pull request by Gouthamve**
Tested this on a Prometheus run. Works well. @fabxc

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/tsdb/138)

### CODE REVIEW

File: index.go [hunk 1]
```diff
@@ -775,10 +788,29 @@ func (r *indexReader) Series(ref uint64, lbls *labels.Labels, chks *[]ChunkMeta)
...
+		ref0 uint64
+	)
+
+	if k == 0 {
+		return nil
+	}
+
+	t0 = d2.varint64()
+	maxt := d2.varint64() + t0
+	ref0 = d2.uvarint64()
```
Comment (fabxc) [inline index.go]: We can just use `:=` here no? No need to specify a `var` block further up.

**Comment by Gouthamve** [comment]
Fixed it. And changed the format to be more efficient.

File: Documentation/format/index.md [hunk 2]
```diff
@@ -94,11 +96,18 @@ After the labels, the number of indexed chunks is encoded, followed by a sequenc
...
+│ │     <uvarint>    │ ┌──────────────────────────┐     │ │
+│ │                  │ │c_i.mint - c_i-1.maxt <uvarint> │ │
 │ │                  │ ├──────────────────────────┤     │ │
-│ │      #chunks     │ │ c_i.maxt <varint>        │     │ │
-│ │     <uvarint>    │ ├──────────────────────────┤ ... │ │
-│ │                  │ │ ref(c_i.data) <uvarint>  │     │ │
+│ │                  │ │ c_i.maxt - c_i.mint <uvarint>  │ │
+│ │                  │ ├──────────────────────────┤ ... │ │
+│ │                  │ │ ref(c_i.data) - ref0 <varint>  │ │
 │ │                  │ └──────────────────────────┘     │ │
```
Comment (fabxc) [inline Documentation/format/index.md]: Can we make the columns all wider so it's nicely in boxes again ;(
I wanted them to be pretty so badly.

File: index.go [hunk 3]
```diff
@@ -292,10 +292,20 @@ func (w *indexWriter) AddSeries(ref uint64, lset labels.Labels, chunks ...ChunkM
...
+		w.buf2.putUvarint64(uint64(c.MaxTime - c.MinTime))
 		w.buf2.putUvarint64(c.Ref)
+		t0 := c.MaxTime
+		ref0 := int64(c.Ref)
+
+		for _, c := range chunks[1:] {
+			w.buf2.putUvarint64(uint64(c.MinTime - t0))
+			w.buf2.putUvarint64(uint64(c.MaxTime - c.MinTime))
+			w.buf2.putVarint64(int64(c.Ref) - ref0)
+			t0 = c.MaxTime
```
Comment (fabxc) [inline index.go]: Any reason for not adapting `ref0` as well?

**Comment by fabxc** [comment]
👍

**Pull request merged**